src/pages/iou/IOUModal.js

```diff
@@ -116,6 +133,14 @@ class IOUModal extends Component {
         if (prevProps.iou.creatingIOUTransaction && !this.props.iou.creatingIOUTransaction && !this.props.iou.error) {
             Navigation.dismissModal();
         }
+
+        if (prevProps.myPersonalDetails.preferredCurrencyCode
```

Comment:
Does this work? I would expect the prevProps and current props to be different instances, so this would always return true, no?

Reply:
In this case, we're comparing two values, so whenever the value changes the previous and current prop are different, so this would return true in that case and false in all other cases.